File: UKG/Nvidiaukg.py
def solve(s):
    n = len(s)
    ct = 0
    flag = False

    for char in s:
        if char == '1':
            ct += 1
        else:
            ct -= 1

        if ct < 0:
            flag = True
            break

    if flag:
        return False
    return ct == 0

def f(s):
    n = len(s)

    for i in range(n):
        if s[i] == '0':
            return i
    return -1

# Read input
s = input()
ans = s
n = len(s)

for i in range(n):
    p = s[:i+1]
    q = s[i+1:]

    if solve(p) and solve(q):
        ko = q + p

        if f(ans) < f(ko):
            ans = ko

print(ans)

File: UKG/Nvidiaukg.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;

int solve(string s){
    int n = s.size();
    int ct = 0;

    int flag = 0;
    for(int i=0;i<n;i++)
    {
        if(s[i]=='1') ct++;
        else ct--;

        if(ct<0){
            flag = 1;
            break;
        }
    }
    if(flag == 1){
        return false;
    }
    return (ct==0);
}

int f(string s){
    int n = s.size();

    for(int i=0;i<n;i++)
    {
        if(s[i]=='0') return i;
    }
    return -1;
}

int main()
{
    string s;
    cin>>s;
    
    string ans = s;

    int n = s.size();
    for(int i=0;i<n;i++)
    {
        string p = s.substr(0,i+1);
        string q = s.substr(i+1,n-i-1);

        if(solve(p) && solve(q)){
            
            string ko = q;
            for(auto x:p) ko += x;

            if(f(ans)<f(ko)){
                ans = ko;
            }
        }
    }
    cout<<ans<<endl;
}